node_cli/core/host.py

```diff
@@ -65,52 +73,71 @@ def fix_url(url):
         return False
 
 
-def get_flask_secret_key():
+def get_flask_secret_key() -> str:
+    """Retrieve Flask secret key from filesystem."""
     secret_key_filepath = os.path.join(NODE_DATA_PATH, 'flask_db_key.txt')
-    with open(secret_key_filepath) as key_file:
-        return key_file.read().strip()
 
+    if not os.path.exists(secret_key_filepath):
+        error_exit(f'Flask secret key file not found at {secret_key_filepath}')
+
+    try:
+        with open(secret_key_filepath, 'r') as key_file:
+            secret_key = key_file.read().strip()
+            return secret_key
+    except (IOError, OSError) as e:
+        error_exit(f'Failed to read Flask secret key: {e}')
+        # Will never reach here, but needed for type checking.
+        return ''
+
+
+def prepare_host(env_filepath: str, env_type: str, allocation: bool = False) -> None:
+    """Initialize SKALE node host environment."""
+    if not env_filepath or not env_type:
+        error_exit('Missing required parameters for host initialization')
 
-def prepare_host(
-    env_filepath: str,
-    env_type: str,
-    allocation: bool = False
-):
-    logger.info('Preparing host started')
-    make_dirs()
-    save_env_params(env_filepath)
+    try:
+        logger.info('Preparing host started')
+        make_dirs()
+        save_env_params(env_filepath)
 
-    if allocation:
-        update_resource_allocation(env_type)
+        if allocation:
+            update_resource_allocation(env_type)
+    except Exception as e:
```

Comment:
It is cleaner to handle base Exception class in one place. It is already handled in main.py. Suggest to raise custom exception type instead and run error_exit in main.py with specific error index.